### ISSUE TITLE
Different results while Decoding  in different machines

### ISSUE THREAD

**Issue by FaridHaziyev** [espnet/espnet#2875]
First of all thank you for this easy to use model, it made it easy to train my model. I have trained my model with espnet1 and tested it worked well. When I tried it to decode in another machine with the given example in the ReadMe (https://colab.research.google.com/github/espnet/notebook/blob/master/espnet2_tts_realtime_demo.ipynb) the result is missing half of the audio. 

When I tried to debug the code,             
if int(sum(probs[-1] >= threshold)) > 0 or idx >= maxlen:

In this line in the decoder.py, it reaches the threshold before actual end of the audio. I am sorry if this is not a bug, but I have tried exact same example as in the tutorial.

**Comment by kan-bayashi** [comment]
This is not a bug.
If you use Tacotron2 or Transformer, the prenet applies dropout in decoding.
See the detail in Tacotron2 paper.

**Comment by FaridHaziyev** [comment]
If you mean this : set_deterministic_pytorch(Namespace(**{"seed":1, "debugmode":1}))
I did that. Actually I use the same code in two different machines and one gives different results than the other. I use this seed parameter in both of them.

**Comment by kan-bayashi** [comment]
I think that is difficult to fix the seed among the machines.
It depends on cuda version or gpus.
https://pytorch.org/docs/stable/notes/randomness.html

**Comment by FaridHaziyev** [comment]
Actually I do not use gpus while decoding

**Comment by kan-bayashi** [comment]
Can you reproduce the deterministic results in the single machine?

**Comment by FaridHaziyev** [comment]
Yes in the same machine I get the same results, each time I run the code

**Comment by kan-bayashi** [comment]
Then, actually I have no idea, it depends on pytorch.
Please check the following page.
https://pytorch.org/docs/stable/notes/randomness.html

**Comment by FaridHaziyev** [comment]
export OMP_NUM_THREADS=1

doing this solved my problem

**Comment by v-yunbin** [comment]
@kan-bayashi for ASR, I met same problem.  i get different result on different machine.
**result on machine1**:
**comand**: `# asr_recog.py --config conf/decode.yaml --ngpu 1 --backend pytorch --batchsize 1 --recog-json offline_test/test/split1utt/data.1.json --result-label offline_test/result/data.1.json --model exp/train_sp_pytorch_train/results/model.acc.best --rnnlm exp/train_rnnlm_pytorch_lm/results/rnnlm.model.best --api v2 `
**result** : `2020-12-28 16:08:12,818 (beam_search:403) INFO: best hypo: 就<space>是<space>那个<space>放映<space>盘<space>发车<space>油门<space>都<space>不用

2020-12-28 16:08:12,819 (asr_utils:852) INFO: groundtruth: 7200 7380 嗯，就是那个方向盘、刹车、油门儿都不用踩的情况下
是吧
2020-12-28 16:08:12,819 (asr_utils:853) INFO: prediction : 就 是 那个 放映 盘 发车 油门 都 不用<eos>
`
**result on machine2**:
**command**:` asr_recog.py --config conf/decode.yaml --ngpu 1 --backend pytorch --batchsize 1 --recog-json offline_test/test/split1utt/data.1.json --result-label offline_test/result/data.1.json --model exp/train_sp_pytorch_train/results/model.acc.best --rnnlm exp/train_rnnlm_pytorch_lm/results/rnnlm.model.best --api v2 `
**result**:`2021-01-19 18:09:38,671 (beam_search:403) INFO: best hypo: 这<space>是<space>对<space>的<space>话<space>有<space>朋友
<space>发<space>吃<space>了<space>有<space>都<space>不用

2021-01-19 18:09:38,671 (asr_utils:852) INFO: groundtruth: 7200 7380 嗯，就是那个方向盘、刹车、油门儿都不用踩的情况下
是吧
2021-01-19 18:09:38,672 (asr_utils:853) INFO: prediction : 这 是 对 的 话 有 朋友 发 吃 了 有 都 不用<eos>
`
